node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/MetricsReporter.java

```diff
@@ -20,22 +29,138 @@
 public class MetricsReporter extends Maintainer {
 
     private final Metric metric;
-    private final HashMap<Flavor, Metric.Context> contextMap = new HashMap<>();
+    private final Orchestrator orchestrator;
+    private final Map<Map<String, String>, Metric.Context> contextMap = new HashMap<>();
 
-    public MetricsReporter(NodeRepository nodeRepository, Metric metric, Duration interval, JobControl jobControl) {
+    public MetricsReporter(NodeRepository nodeRepository,
+                           Metric metric,
+                           Orchestrator orchestrator,
+                           Duration interval,
+                           JobControl jobControl) {
         super(nodeRepository, interval, jobControl);
         this.metric = metric;
+        this.orchestrator = orchestrator;
     }
 
     @Override
     public void maintain() {
+        List<Node> nodes = nodeRepository().getNodes();
+        nodes.forEach(this::updateNodeMetrics);
+        updateStateMetrics(nodes);
+        updateDockerMetrics(nodes);
+    }
+
+    private void updateNodeMetrics(Node node) {
+        // Dimensions automatically added: host, vespaVersion, zone, role, and colo.
```

Comment:
What does 'automatically' mean here? If this is done elsewhere, this comment will become outdated when the other code changes.

Reply:
Let me remove the comment.